libminifi/src/core/ProcessSession.cpp

```diff
@@ -469,6 +470,14 @@ void ProcessSession::import(std::string source, const std::shared_ptr<core::Flow
           invalidWrite = true;
         }
       }
+
+      io::BaseStream *stream = baseStream.get();
+      std::unique_ptr<io::CRCStream<io::BaseStream>> crcStream;
+      if (crc) {
+        crcStream = utils::make_unique<io::CRCStream<io::BaseStream>>(stream, *crc);
+        stream = crcStream.get();
+      }
```

Comment:
I agree this is a bit nasty.  Let me think of a better way, too, and then we'll discuss it.

Reply:
I thought a bit about this. There is a function similar to `ProcessSession::import` called `ProcessSession::importFrom` which takes an input stream instead of a file name. It doesn't support FlowFile delimiters, though.

My idea is to extend `ProcessSession::importFrom` to support delimiters and use that with a `CRCStream` over a `FileStream`. If I understand correctly, we include every byte that we processed in the checksum, so the stream composition wouldn't change behavior, but correct me if I'm wrong.